File: packages/core/src/config/paths.ts
import { homedir } from 'node:os'
import { join, dirname } from 'node:path'

const appName = 'nekocode'

function xdgConfig(): string {
  return process.env['XDG_CONFIG_HOME'] ?? join(homedir(), '.config')
}
function xdgData(): string {
  return process.env['XDG_DATA_HOME'] ?? join(homedir(), '.local', 'share')
}
function xdgCache(): string {
  return process.env['XDG_CACHE_HOME'] ?? join(homedir(), '.cache')
}
function xdgState(): string {
  return process.env['XDG_STATE_HOME'] ?? join(homedir(), '.local', 'state')
}

export const paths = {
  /** ~/.config/nekocode/settings.jsonc  (or NEKOCODE_CONFIG override) */
  config: process.env['NEKOCODE_CONFIG']
    ?? join(xdgConfig(), appName, 'settings.jsonc'),

  /** ~/.local/share/nekocode/sessions/ */
  sessions: join(xdgData(), appName, 'sessions'),

  /** ~/.cache/nekocode/ */
  cache: join(xdgCache(), appName),

  /** ~/.local/state/nekocode/nekocode.log */
  log: join(xdgState(), appName, 'nekocode.log'),

  /** ~/.local/share/nekocode/skills/ */
  skills: join(xdgData(), appName, 'skills'),

  get configDir(): string {
    return dirname(this.config)
  },

  /** .nekocode/settings.jsonc — project-level config (commit this) */
  projectConfig(cwd: string): string {
    return join(cwd, '.nekocode', 'settings.jsonc')
  },

  /** .nekocode/settings.local.jsonc — local overrides (add to .gitignore) */
  projectLocalConfig(cwd: string): string {
    return join(cwd, '.nekocode', 'settings.local.jsonc')
  },
}

File: crates/neko-core/src/session/paths.rs
use std::path::PathBuf;

fn xdg_config() -> PathBuf {
    std::env::var("XDG_CONFIG_HOME")
        .map(PathBuf::from)
        .unwrap_or_else(|_| dirs::home_dir().unwrap_or_default().join(".config"))
}

fn xdg_data() -> PathBuf {
    std::env::var("XDG_DATA_HOME")
        .map(PathBuf::from)
        .unwrap_or_else(|_| dirs::home_dir().unwrap_or_default().join(".local").join("share"))
}

fn xdg_cache() -> PathBuf {
    std::env::var("XDG_CACHE_HOME")
        .map(PathBuf::from)
        .unwrap_or_else(|_| dirs::home_dir().unwrap_or_default().join(".cache"))
}

fn xdg_state() -> PathBuf {
    std::env::var("XDG_STATE_HOME")
        .map(PathBuf::from)
        .unwrap_or_else(|_| dirs::home_dir().unwrap_or_default().join(".local").join("state"))
}

const APP: &str = "neko";

/// ~/.config/neko/settings.jsonc
pub fn config_path() -> PathBuf {
    std::env::var("NEKO_CONFIG")
        .map(PathBuf::from)
        .unwrap_or_else(|_| xdg_config().join(APP).join("settings.jsonc"))
}

/// ~/.config/neko/
pub fn config_dir() -> PathBuf {
    config_path().parent().unwrap().to_path_buf()
}

/// ~/.local/share/neko/sessions/
pub fn sessions_dir() -> PathBuf {
    xdg_data().join(APP).join("sessions")
}

/// ~/.cache/neko/
pub fn cache_dir() -> PathBuf {
    xdg_cache().join(APP)
}

/// ~/.local/state/neko/
pub fn state_dir() -> PathBuf {
    xdg_state().join(APP)
}

/// ~/.local/state/neko/neko.log
pub fn log_path() -> PathBuf {
    state_dir().join("neko.log")
}

/// ~/.local/state/neko/history
pub fn history_path() -> PathBuf {
    state_dir().join("history")
}

/// ~/.local/share/neko/skills/
pub fn skills_dir() -> PathBuf {
    xdg_data().join(APP).join("skills")
}
